### DISCUSSION TITLE
Mirth in Azure

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5773] [Q&A]
We are embarking on a project to move our Mirth instances to Azure. Would appreciate any recommendations, experiences, things to consider or gotcha's. Any problems you might have run into. Thanks

**Comment by pacmano1** [comment]
Use AWS. 

Sorry, it's like a wet paint sign or when the waiter says "the plate is hot".

**Comment by rutledj** [comment]
Any reason AWS is better than Azure?

**Reply by pacmano1** [reply]
It was a joke. They are competitive with each other.

**Comment by jonbartels** [comment]
My employer is currently making this migration. I am NOT a devops person at all. My perception as an application engineer is that this migration from AWS to Azure is made easier because the company has generally standardized on k8s. So with docker images tuned the way we want them and helm charts and terraform, it is in theory deployable or easily migratable to almost any cloud vendor.

Some decisions when running in Azure are what common apps to standardize on. Some examples:

* I prefer postgres but in Azure there is a bias towards SQL Server. What does your company want to do? Is there a cost savings for SQL Server over Postgres in Azure?
* AD or a different IdP? Same as above really, Azure will probably bias towards ActiveDirectory. 

This migration has also been a chance to catch up on some tech debt. My team is defining our Mirth images the way we want, not the way some devops engineer guessed (I say that lovingly and respectfully!). We're using more environment variables. We don't have persistent storage beyond the database. We have our plugins set how *I* want them as a Mirth expert.

I don't know how much that helps. Anecdotes != data.

**Comment by rutledj** [comment]
I think they are leaning toward sql server as they already have the licensing for that in Azure. Not sure about AD. I don't think there is anyone on the team that even knows how to set up the VM or Mirth image.  I suspect they will be utilizing a 3rd party to help with the architecture and initial setup.

**Comment by pacmano1** [comment]
Regardless of your cloud vendor - the infrastructure behind mirth and familiarity with designing and troubleshooting it should be paramount.  For example, if you need IPSec tunnels - who will build those?   If you need public facing API endpoints, what tech will you choose?   WAFs, IDS/IPS, cloud posture management and so on.    The list can be very long and it is not a "set it and forget it".  You need resources assiged to maintaining your infrastructure like you would for an in house deployment.
